collect.py

```diff
@@ -223,15 +229,21 @@ def run_tests(firefox_driver, chrome_driver):
 
     for bug in bugs:
         try:
-            # Assume that if we generated the main file, we also generated the one with
-            # the sequence of operations (TODO: don't assume, check!)
-            # TODO: If only Chrome is missing, don't regenerate Firefox too, but read the
-            # sequence of operations from the files.
-            if not os.path.exists('data/%d_firefox.png' % bug['id']) or\
-               not os.path.exists('data/%d_chrome.png' % bug['id']):
+            if not os.path.exists('data/%d.txt' % bug['id']) or \
+                    not os.path.exists('data/%d_firefox.png' % bug['id']) or \
+                    not os.path.exists('data/%d_chrome.png' % bug['id']):
+                sequence = run_test(bug, 'firefox', firefox_driver)
+                run_test(bug, 'chrome', chrome_driver, sequence)
+                with open("data/%d.txt" % bug['id'], 'w') as f:
+                    for element in sequence:
+                        f.write(json.dumps(element) + '\n')
+            sequence = read_sequence(bug['id'])
+            number_of_ff_scr = len(glob.glob('data/%d_*_firefox.png' % bug['id']))
+            number_of_ch_scr = len(glob.glob('data/%d_*_chrome.png' % bug['id']))
+            if len(sequence) != number_of_ff_scr or \
```

Comment:
I'm not sure this takes care of the case "we haven't generated any item of the sequence for Firefox". In this case, the sequence file is empty, so len(sequence) is 0 and number_of_ff_scr is 0 too.